crates/core/src/utils.rs

```diff
@@ -130,8 +133,13 @@ pub fn hash_content<R: Read>(content: &mut R) -> [u8; 32] {
 /// location can be checked.
 pub fn hash_folder(folder_path: impl AsRef<Path>) -> Result<IntegrityChecksum, std::io::Error> {
     // a list of hashes, one for each DirEntry
-    let all_hashes = Arc::new(Mutex::new(Vec::with_capacity(100)));
+    //    let all_hashes = Arc::new(Mutex::new(Vec::with_capacity(100)));
     let root_path = Arc::new(dunce::canonicalize(folder_path.as_ref())?);
+
+    let (tx, rx) = std::sync::mpsc::channel::<[u8; 32]>();
+    let tx = Arc::new(tx);
+
+    let mut hashes = Vec::new();
```

Comment:
This vec allocation can be moved to after the `walker.run`, just before we loop over `rx.recv()`